src/uniswapMappings/uniswapPools.ts

```diff
@@ -0,0 +1,143 @@
+/* eslint-disable prefer-const */
+import { Bundle, UniswapPool, Token } from '../../generated/schema'
+import { BigDecimal, BigInt } from '@graphprotocol/graph-ts'
+import {
+  Burn as BurnEvent,
+  Initialize,
+  Mint as MintEvent,
+  Swap as SwapEvent
+} from '../../generated/templates/Pool/Pool'
+import { convertTokenToDecimal } from '../utils'
+import { ZERO_BD } from '../utils/constants'
+import { findEthPerToken, getEthPriceInUSD, sqrtPriceX96ToTokenPrices } from '../utils/pricing'
+
+export function handleInitialize(event: Initialize): void {
+  let pool = UniswapPool.load(event.address.toHexString())
+  pool.tick = BigInt.fromI32(event.params.tick)
+  // update token prices
+  let token0 = Token.load(pool.token0)
+  let token1 = Token.load(pool.token1)
+
+  // update ETH price now that prices could have changed
+  let bundle = Bundle.load('1')
+  bundle.ethPriceUSD = getEthPriceInUSD()
+  bundle.save()
+
+  // update token prices
+  token0.derivedETH = findEthPerToken(token0 as Token)
+  token1.derivedETH = findEthPerToken(token1 as Token)
+  token0.derivedUSD = token0.derivedETH.times(bundle.ethPriceUSD)
+  token1.derivedUSD = token1.derivedETH.times(bundle.ethPriceUSD)
+  token0.save()
+  token1.save()
+}
+
+export function handleMint(event: MintEvent): void {
```

Comment:
do we handle mint? why is this relevant?

Reply:
i guss is for calculating TVL. Will we need this?

Reply:
I see its used for the algo that finds the price, so NVM

Reply:
yehap it's changing pool's tvl. we need to keep it for that.